components/mosquitto/examples/serverless_mqtt/components/libjuice/CMakeLists.txt

```diff
@@ -0,0 +1,44 @@
+set(LIBJUICE_VERSION "73785387eafe15c02b6a210edb10f722474e8e14")
+set(LIBJUICE_URL "https://github.com/paullouisageneau/libjuice/archive/${LIBJUICE_VERSION}.zip")
+
+set(libjuice_dir ${CMAKE_BINARY_DIR}/libjuice/libjuice-${LIBJUICE_VERSION})
+
+# Fetch the library
+if(NOT EXISTS ${libjuice_dir})
```

Comment:
I guess this could be simplified by using cmake fetch content and making this a wrapper component.

Reply:
actually I spent quite some time on this trying to use more `cmake`-ky way of including subproject, have tried both `ExternalProject` and `FetchContent` modules, but wasn't able to make them work with IDF cmake wrapper.
this "manual" download was the only thing that worked for me.